### PR TITLE
[Data] Additional args for Data + Train benchmark

### PR DESCRIPTION
**Pull request by scottjlee**
## Why are these changes needed?

As a followup to https://github.com/ray-project/ray/pull/37624, add the following additional parameters for the multi-node training benchmark:
- File type (image, parquet)
- local shuffle buffer size
- preserve_order (train config)
- increases default # epochs to 10

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottjlee** [comment]
Release test outputs:
- `read_images_train_4_cpu`:
```
{"read_image_train_4workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 5.410256495549528}}, "success": 1}
```

- `read_images_train_16_cpu`:
```
{"read_image_train_16workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 0.942105386854395}}, "success": 1}
```
- `read_images_train_16_cpu_preserve_order`:
```
{"read_image_train_16workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 0.939200517950852}}, "success": 1}
```

- `read_parquet_train_4_cpu`:
```
{"read_parquet_train_4workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 117.9580262429022}}, "success": 1}
```

- `read_parquet_train_16_cpu`:
```
{"read_parquet_train_16workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 28.508951310541576}}, "success": 1}
```

**Comment by stephanie-wang** [comment]
> Release test outputs:
> 
>     * `read_images_train_4_cpu`:
> 
> 
> ```
> {"read_image_train_4workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 5.410256495549528}}, "success": 1}
> ```
> 
>     * `read_images_train_16_cpu`:
> 
> 
> ```
> {"read_image_train_16workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 0.942105386854395}}, "success": 1}
> ```
> 
>     * `read_images_train_16_cpu_preserve_order`:
> 
> 
> ```
> {"read_image_train_16workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 0.939200517950852}}, "success": 1}
> ```
> 
>     * `read_parquet_train_4_cpu`:
> 
> 
> ```
> {"read_parquet_train_4workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 117.9580262429022}}, "success": 1}
> ```
> 
>     * `read_parquet_train_16_cpu`:
> 
> 
> ```
> {"read_parquet_train_16workers": {"ray_TorchTrainer_fit": {"THROUGHPUT": 28.508951310541576}}, "success": 1}
> ```

The changes look good, but aren't the throughputs here a bit odd? I thought the throughput should go up instead of down for 4 -> 16 CPUs?

File: release/release_tests.yaml [hunk 1]
```diff
@@ -6401,8 +6401,86 @@
...
+  variations:
+    - __suffix__: aws
+    - __suffix__: gce
+      env: gce
+      frequency: manual
+      cluster:
+        cluster_env: app_config.yaml
+        cluster_compute: ../../air_tests/air_benchmarks/compute_gpu_4x4_gce.yaml
+
+- name: read_images_train_16_cpu_preserve_order
```
Comment (c21) [inline release/release_tests.yaml:6416]: `cpu` -> `gpu` right?

File: release/release_tests.yaml [hunk 2]
```diff
@@ -6401,8 +6401,86 @@
...
+  variations:
+    - __suffix__: aws
+    - __suffix__: gce
+      env: gce
+      frequency: manual
+      cluster:
+        cluster_env: app_config.yaml
+        cluster_compute: ../../air_tests/air_benchmarks/compute_gpu_4x4_gce.yaml
+
+- name: read_parquet_train_4_cpu
```
Comment (c21) [inline release/release_tests.yaml:6442]: ditto

File: release/release_tests.yaml [hunk 3]
```diff
@@ -6401,8 +6401,86 @@
...
+  variations:
+    - __suffix__: aws
+    - __suffix__: gce
+      env: gce
+      frequency: manual
+      cluster:
+        cluster_env: app_config.yaml
+        cluster_compute: ../../air_tests/air_benchmarks/compute_gpu_2x2_gce.yaml
+
+- name: read_parquet_train_16_cpu
```
Comment (c21) [inline release/release_tests.yaml:6468]: ditto

File: release/release_tests.yaml (hunk 1 above)
Comment (c21) [reply release/release_tests.yaml:6416]: If yes, can you also update line 6364 `- name: read_images_train_4_cpu`

File: release/nightly_tests/dataset/multi_node_train_benchmark.py [hunk 4]
```diff
@@ -102,38 +132,46 @@ def train_loop_per_worker():
...
+        scaling_config=ScalingConfig(num_workers=args.num_workers),
+        dataset_config=ray.train.DataConfig(
+            execution_options=options,
+        ),
     )
     result = torch_trainer.fit()
 
     # Report the throughput of the last training epoch.
-    metrics["ray.TorchTrainer.fit"] = list(result.metrics_dataframe["tput"])[-1]
+    metrics["ray_TorchTrainer_fit"] = list(result.metrics_dataframe["tput"])[-1]
```
Comment (c21) [inline release/nightly_tests/dataset/multi_node_train_benchmark.py:160]: how this `metrics_dataframe` being aggregated across training workers? is this the sum up of individual `tput` from each worker?

**Review by c21 — APPROVED** [review]
LG, let's debug the throughput performance separately.

**Pull request merged**